### PR TITLE
docs: improve documentation around fileMatch being additive

### PR DESCRIPTION
**Pull request by mschoettle**
## Changes

Improve the documentation around `fileMatch` that configuring it is additive and mention `ignorePaths` and `includePaths` to exclude/include specific paths.

## Context

See the discussion in #20080.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: tools/docs/manager.ts [hunk 1]
```diff
@@ -152,7 +152,7 @@ sidebar_label: ${displayName}
           md += '\n```\n\n';
         }
       }
-      md += `For details on how to extend a manager's \`fileMatch\` value, please follow [this link](/modules/manager/#file-matching).\n\n`;
```
Comment (mschoettle) [inline tools/docs/manager.ts:155]: This might not be necessary given the other changes.

**Review by rarkins — CHANGES_REQUESTED** [review]
One sentence per line

File: docs/usage/configuration-options.md [hunk 2]
```diff
@@ -821,7 +821,10 @@ Renovate can fetch release notes when they are hosted on one of these platforms:
 
 ## fileMatch
 
-`fileMatch` is used by Renovate to know which files in a repository to parse and extract, and it is possible to override the default values to customize for your project's needs.
+`fileMatch` is used by Renovate to know which files in a repository to parse and extract.
+`fileMatch` is a mergeable field, i.e., patterns can be added to the default patterns of a manager.
```
Comment (rarkins) [inline docs/usage/configuration-options.md:825]:
```suggestion
`fileMatch` patterns in user config are added to the default values and do not replace them.
```

File: docs/usage/configuration-options.md [hunk 3]
```diff
@@ -821,7 +821,10 @@ Renovate can fetch release notes when they are hosted on one of these platforms:
...
+The default patterns cannot be removed.
+If you need to include or exclude specific paths use the `ignorePaths` or `includePaths` configuration options.
```
Comment (rarkins) [inline docs/usage/configuration-options.md:827]:
```suggestion
The default `fileMatch` patterns cannot be removed, so if you need to include or exclude specific paths then use the `ignorePaths` or `includePaths` configuration options.
```

**Review by rarkins — CHANGES_REQUESTED** [review]

File: tools/docs/manager.ts (hunk 1 above)
Comment (rarkins) [reply tools/docs/manager.ts:155]: I think let's revert this change

File: tools/docs/manager.ts [hunk 4]
```diff
@@ -152,7 +152,7 @@ sidebar_label: ${displayName}
...
+      md += `For details on how to extend a manager's \`fileMatch\` value, please follow [this link](/modules/manager/#file-matching). To have the manager ignore specific paths use the [\`ignorePaths\`](/configuration-options/#ignorepaths) configuration option.\n\n`;
```
Comment (rarkins) [inline tools/docs/manager.ts:155]:
```suggestion
      md += `For details on how to extend a manager's \`fileMatch\` value, please follow [this link](/modules/manager/#file-matching).\n\n`;
```

**Review by rarkins — APPROVED** [review]

**Pull request merged**